src/Compilers/CSharp/Portable/Syntax/SyntaxExtensions.cs

```diff
@@ -70,15 +70,18 @@ public static SyntaxToken NormalizeWhitespace(this SyntaxToken token, string ind
         /// </summary>
         internal static TypeSyntax Type(this DeclarationExpressionSyntax self)
         {
-            return self.Declaration.Type;
+            var component = (TypedVariableComponentSyntax)self.Declaration;
```

Comment:
Oh... this is internal.  I dno't care then.  If these become public (whihc seems nice), then i'd want the checks.